--- libpcap-0.9.6/VERSION.shared	2006-09-19 21:27:07.000000000 +0200
+++ libpcap-0.9.6/VERSION	2007-06-19 13:30:34.000000000 +0200
@@ -1 +1 @@
-0.9.5
+0.9.6
--- libpcap-0.9.6/Makefile.in.shared	2006-09-25 20:06:00.000000000 +0200
+++ libpcap-0.9.6/Makefile.in	2007-06-19 13:30:15.000000000 +0200
@@ -22,6 +22,7 @@
 #
 # Various configurable paths (remember to edit Makefile.in, not Makefile)
 #
+VERSION := $(shell cat VERSION)
 
 # Top level hierarchy
 prefix = @prefix@
@@ -96,9 +97,14 @@ TAGHDR = \
 TAGFILES = \
 	$(SRC) $(HDR) $(TAGHDR)
 
-CLEANFILES = $(OBJ) libpcap.a $(GENSRC) $(GENHDR) lex.yy.c
 
-all: libpcap.a
+
+CLEANFILES = $(OBJ) libpcap.a $(GENSRC) $(GENHDR) lex.yy.c libpcap.so.$(VERSION)
+
+all: libpcap.a libpcap.so.$(VERSION)
+
+libpcap.so.$(VERSION): $(OBJ)
+	gcc -shared -Wl,-soname -Wl,libpcap.so.$(SOVERSION) -o libpcap.so.$(VERSION) $(OBJ) 
 
 libpcap.a: $(OBJ)
 	@rm -f $@
@@ -171,10 +177,13 @@ bpf_filter.c: $(srcdir)/bpf/net/bpf_filt
 bpf_filter.o: bpf_filter.c
 	$(CC) $(CFLAGS) -c bpf_filter.c
 
-install: libpcap.a 
+install: libpcap.a libpcap.so.$(VERSION)
 	[ -d $(DESTDIR)$(libdir) ] || \
 	    (mkdir -p $(DESTDIR)$(libdir); chmod 755 $(DESTDIR)$(libdir))
 	$(INSTALL_DATA) libpcap.a $(DESTDIR)$(libdir)/libpcap.a
+	$(INSTALL_PROGRAM) libpcap.so.$(VERSION)  $(DESTDIR)$(libdir)/libpcap.so.$(VERSION)
+	ln -sf libpcap.so.$(VERSION) $(DESTDIR)$(libdir)/libpcap.so.$(SOVERSION)
+	ln -sf libpcap.so.$(SOVERSION) $(DESTDIR)$(libdir)/libpcap.so
 	$(RANLIB) $(DESTDIR)$(libdir)/libpcap.a
 	[ -d $(DESTDIR)$(includedir) ] || \
 	    (mkdir -p $(DESTDIR)$(includedir); chmod 755 $(DESTDIR)$(includedir))
