diff -up libpcap/Makefile.in.oldsoname libpcap/Makefile.in
--- libpcap/Makefile.in.oldsoname	2009-07-17 00:10:42.000000000 +0200
+++ libpcap/Makefile.in	2009-07-22 15:53:32.000000000 +0200
@@ -328,6 +328,8 @@ libpcap.so: $(OBJ)
 	MAJOR_VER=`sed 's/\([0-9][0-9]*\)\..*/\1/' $(srcdir)/VERSION`; \
 	@V_SHLIB_CMD@ @V_SHLIB_OPT@ @V_SONAME_OPT@$@.$$MAJOR_VER $(LDFLAGS) \
 	    -o $@.$$VER $(OBJ) $(ADDLOBJS) $(LIBS)
+	@V_SHLIB_CMD@ @V_SHLIB_OPT@ @V_SONAME_OPT@$@.0.9 $(LDFLAGS) \
+	    -o $@.0.9.9 $(OBJ) $(ADDLOBJS) $(LIBS)
 
 #
 # The following rule succeeds, but the result is untested.
@@ -542,6 +544,8 @@ install-shared-so: libpcap.so
 	VER=`cat $(srcdir)/VERSION`; \
 	MAJOR_VER=`sed 's/\([0-9][0-9]*\)\..*/\1/' $(srcdir)/VERSION`; \
 	$(INSTALL_PROGRAM) libpcap.so.$$VER $(DESTDIR)$(libdir)/libpcap.so.$$VER; \
+	$(INSTALL_PROGRAM) libpcap.so.0.9.9 $(DESTDIR)$(libdir)/libpcap.so.0.9.9; \
+	ln -sf libpcap.so.0.9.9 $(DESTDIR)$(libdir)/libpcap.so.0.9; \
 	ln -sf libpcap.so.$$VER $(DESTDIR)$(libdir)/libpcap.so.$$MAJOR_VER; \
 	ln -sf libpcap.so.$$MAJOR_VER $(DESTDIR)$(libdir)/libpcap.so
 install-shared-dylib: libpcap.dylib
