diff -up libpcap/pcap-linux.c.oneshot libpcap/pcap-linux.c
--- libpcap/pcap-linux.c.oneshot	2009-07-17 00:10:42.000000000 +0200
+++ libpcap/pcap-linux.c	2009-07-22 14:30:52.000000000 +0200
@@ -2744,7 +2744,7 @@ static void
 pcap_oneshot_mmap(u_char *user, const struct pcap_pkthdr *h,
     const u_char *bytes)
 {
-	struct pkt_for_oneshot *sp = (struct pkt_for_oneshot *)user;
+	struct oneshot_userdata *sp = (struct oneshot_userdata *)user;
 	bpf_u_int32 copylen;
 
 	*sp->hdr = *h;
